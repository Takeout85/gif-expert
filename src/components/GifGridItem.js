import React from "react";
import propTypes from "prop-types";
import "animate.css";

const GifGridItem = ({ id, title, url }) => {
  return (
    <div key={id} className="card animate__animated animate__backInLeft">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  id: propTypes.string.isRequired,
  title: propTypes.string.isRequired,
  url: propTypes.string.isRequired,
};

export default GifGridItem;
