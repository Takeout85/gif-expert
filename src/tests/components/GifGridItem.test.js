import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Tests on the <GifGridItem /> component", () => {
  const title = "Title";
  const url = "http://localhost/something.jpg";
  const id = "123";
  const wrapper = shallow(<GifGridItem title={title} url={url} id={id} />);

  test("should show the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should show the component correctly with a title", () => {
    expect(wrapper.find("p").text().trim()).toBe(title);
  });

  test("should have the image equal to the url and alt of the props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("should have the animate__animated and animate__backInLeft classes", () => {
    const div = wrapper.find("div");
    expect(div.hasClass("animate__animated")).toBe(true);
    expect(div.hasClass("animate__backInLeft")).toBe(true);
  });
});
