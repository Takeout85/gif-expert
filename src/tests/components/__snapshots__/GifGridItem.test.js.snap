// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on the <GifGridItem /> component should show the component correctly 1`] = `
<div
  className="card animate__animated animate__backInLeft"
  key="123"
>
  <img
    alt="Title"
    src="http://localhost/something.jpg"
  />
  <p>
    Title
  </p>
</div>
`;
